firewall/interception.go

```diff
@@ -188,6 +80,90 @@ func handlePacket(pkt packet.Packet) {
 	conn.HandlePacket(pkt)
 }
 
+// fastTrackedPermit quickly permits certain network criticial or internal connections.
+func fastTrackedPermit(pkt packet.Packet) (handled bool) {
+	meta := pkt.Info()
+
+	switch meta.Protocol {
+	case packet.ICMP:
+		// Always permit ICMP.
+		log.Debugf("accepting ICMP: %s", pkt)
+		_ = pkt.PermanentAccept()
+		return true
+
+	case packet.ICMPv6:
+		// Always permit ICMPv6.
+		log.Debugf("accepting ICMPv6: %s", pkt)
+		_ = pkt.PermanentAccept()
+		return true
+
+	case packet.UDP, packet.TCP:
+		switch meta.DstPort {
+
+		case 67, 68, 546, 547:
+			// Always allow DHCP, DHCPv6.
+
+			// DHCP and DHCPv6 must be UDP.
+			if meta.Protocol != packet.UDP {
+				return false
+			}
+
+			// DHCP is only valid in local network scopes.
+			switch netutils.ClassifyIP(meta.Dst) {
+			case netutils.HostLocal, netutils.LinkLocal, netutils.SiteLocal, netutils.LocalMulticast:
+			default:
+				return false
+			}
+
+			// Log and permit.
+			log.Debugf("accepting DHCP: %s", pkt)
+			_ = pkt.PermanentAccept()
+			return true
+
+		case apiPort:
+			// Always allow direct access to the Portmaster API.
+
+			// Check if the api port is even set.
+			if !apiPortSet {
+				return false
+			}
+
+			// Portmaster API must be TCP
+			if meta.Protocol != packet.TCP {
+				return false
+			}
+
+			fallthrough
+		case 53:
+			// Always allow direct local access to own services.
+			// DNS is both UDP and TCP.
+
+			// Only allow to own IPs.
+			dstIsMe, err := netenv.IsMyIP(meta.Dst)
+			if err != nil {
+				log.Warningf("filter: failed to check if IP is local: %s", err)
+			}
+			if !dstIsMe {
+				return false
+			}
+
+			// Log and permit.
+			switch meta.DstPort {
+			case 53:
+				log.Debugf("accepting local dns: %s", pkt)
+			case apiPort:
+				log.Debugf("accepting api connection: %s", pkt)
+			default:
+				log.Debugf("accepting connection to internal service: %s", pkt)
```

Comment:
This once cannot happen so I'd rather drop it instead of accepting the connection

Reply:
That is true, I viewed it more as a defensive thing in case you forget to add the port to this switch case too. But it is cleaner to just return `false` and have the normal procedure handle it.